crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -38,11 +38,7 @@ pub enum Error {
 pub(crate) fn do_finalize_domain_current_epoch<T: Config>(
     domain_id: DomainId,
     domain_block_number: T::DomainNumber,
-) -> Result<bool, Error> {
-    if domain_block_number % T::StakeEpochDuration::get() != Zero::zero() {
```

Comment:
I would prefer to keep this here than outer because all the logic to the epoch IMHO should be in staking_epoch. Any reason why this change is required?

Reply:
Because we want to call `do_finalize_domain_current_epoch` for the genesis domain with domain block `#1`  (in `on_initialize`), if this condition is kept inside this func then the genesis domain will need to wait for more domain blocks before its operator is finalized.

I'm also fine with removing these changes but calling `do_finalize_domain_current_epoch` for the genesis domain with domain block `#0` is a bit weird though.